inc/Engine/CDN/RocketCDN/DataManagerSubscriber.php

```diff
@@ -315,4 +336,35 @@ private function schedule_subscription_check( $subscription ) {
 			wp_schedule_single_event( $timestamp, self::CRON_EVENT );
 		}
 	}
+
+	/**
+	 * Upgrade callback.
+	 *
+	 * @param string $new_version Plugin new version.
+	 * @param string $old_version Plugin old version.
+	 * @return void
+	 */
+	public function refresh_cdn_cname( $new_version, $old_version ): void {
+		if ( version_compare( $old_version, '3.17.3', '>=' ) ) {
+			return;
+		}
+
+		delete_transient( 'rocketcdn_status' );
+
+		$new_subscription_data = $this->api_client->get_subscription_data();
+		if ( ! $new_subscription_data['is_active'] || empty( $new_subscription_data['cdn_url'] ) ) {
+			return;
+		}
+
+		$cdn_cnames = $this->options->get( 'cdn_cnames', [] );
+		if ( empty( $cdn_cnames[0] ) || $cdn_cnames[0] === $new_subscription_data['cdn_url'] ) {
+			return;
+		}
+
+		$this->options->set( 'rocketcdn_old_url', $cdn_cnames[0] );
```

Comment:
This probably should be options_api instead of options

Reply:
This should not be part of the `wp_rocket_settings` array, it's not an option of the plugin, and it would be removed the next time the settings are saved (because it's not part of the settings page form that is submitted)

Reply:
u r awesome @remyperona 
fixing that now